adminSiteClient/GrapherConfigGridEditor.tsx

```diff
@@ -244,27 +301,193 @@ export class GrapherConfigGridEditor extends React.Component<GrapherConfigGridEd
-        if (richDataRows === undefined) return
+        const { selectedRowContent } = this
+        if (selectedRowContent === undefined) return
 
         // Get the grapherConfig of the currently selected row and then
         // merge it with the necessary partial information (e.g. variableId field)
         // to get a config that actually works in all cases
-        const grapherConfig = richDataRows[row].config
-        const finalConfigLayer = config.finalVariableLayerModificationFn(
-            richDataRows[row].id
+        const grapherConfig = selectedRowContent.config
+        const finalConfigLayer = this.config.finalVariableLayerModificationFn(
+            selectedRowContent.id
         )
 
         const mergedConfig = merge(grapherConfig, finalConfigLayer)
         this.loadGrapherJson(mergedConfig)
     }
 
+    @computed private get columnDataSource(): ColumnDataSource | undefined {
+        const { selectedRowContent, columnDataSources, selectedColumn } = this
+        if (selectedRowContent === undefined) return undefined
+        if (
+            selectedColumn === undefined ||
+            selectedColumn >= columnDataSources.length
+        ) {
+            return undefined
+        }
+        return columnDataSources[selectedColumn]
+    }
+
+    @computed private get currentColumnFieldDesription():
+        | FieldDescription
+        | undefined {
+        const { columnDataSource } = this
+
+        return match(columnDataSource)
+            .with(
+                { kind: ColumnDataSourceType.FieldDescription },
+                (source) => source.description
+            )
+            .otherwise(() => undefined)
+    }
+
+    @action.bound
+    private onRichTextEditorChange(
+        editor: codemirror.Editor,
+        data: codemirror.EditorChange,
+        value: string
+    ) {
+        console.log(
+            "has uncommited changes?",
+            this.hasUncommitedRichEditorChanges
+        )
+        if (data.origin !== undefined) {
+            console.log({ data })
+            // origin seems to be +input when editing and undefined when we change the value programmatically
+            const { currentColumnFieldDesription, grapher } = this
+            if (currentColumnFieldDesription === undefined) return
+            this.hasUncommitedRichEditorChanges = true
+            const pointer = jsonpointer.parse(
+                currentColumnFieldDesription.pointer
+            ) as string[]
+            // Here we are setting the target column field directly at the grapher so the
+            // preview is updated. When the user clicks the save button we'll then check
+            // the value on grapher vs the value on the richDataRow and perform a proper
+            // update using doAction like we do on the grid editor commits.
+            setValueRecursiveInplace(grapher, pointer, value)
+        }
+    }
+
+    @action.bound
+    commitRichEditorChanges(performCommit: boolean) {
```

Comment:
I was confused what `performCommit` does (in a commit method), and was surprised that `performCommit = false` is a rollback.
You should clarify that a bit, or at least add a comment.

Reply:
Added two helper functions without the boolean flag to make this clearer